File: bbc/libs/BRXSDK_Bcad_V23_2_03/samples/CsBrxMgd/CsBrxMgd/BrxMgdForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Runtime.InteropServices;


//ODA
using Teigha.Runtime;
using Teigha.DatabaseServices;
using Teigha.Geometry;

//Bricsys
using Bricscad.ApplicationServices;

//com
using BricscadDb;
using BricscadApp;

// alias
using _AcAp = Bricscad.ApplicationServices;
using _AcDb = Teigha.DatabaseServices;
using _AcGe = Teigha.Geometry;
using _AcEd = Bricscad.EditorInput;

namespace CsBrxMgd
{
    public partial class BrxMgdForm : Form
    {
        AcadDocument m_document;
        public BrxMgdForm(AcadDocument document)
        {
            if (document == null)
                throw new System.ArgumentNullException("document == null");
            InitializeComponent();
            m_document = document;
        }

        private void select_Click(object sender, EventArgs e)
        {
            _AcEd.Editor editor = _AcAp.Application.DocumentManager.MdiActiveDocument.Editor;
            try
            {
                _AcEd.PromptPointOptions pntOpt = new _AcEd.PromptPointOptions("Get First Point");
                _AcEd.PromptPointResult pntRes = editor.GetPoint(pntOpt);

                if (pntRes.Status != Bricscad.EditorInput.PromptStatus.OK)
                    throw new System.Exception("_AcEd.Editor.GetPoint failed");

                _AcEd.PromptDistanceOptions distOpt = new _AcEd.PromptDistanceOptions("Get Next Point");
                distOpt.UseBasePoint = true;
                distOpt.BasePoint = pntRes.Value;
                _AcEd.PromptDoubleResult distRes = editor.GetDistance(distOpt);

                if (distRes.Status != Bricscad.EditorInput.PromptStatus.OK)
                    throw new System.Exception("_AcEd.Editor.GetDistance failed");

                System.Windows.Forms.MessageBox.Show(distRes.ToString());
            }
            catch (System.Exception ex)
            {
                System.Windows.Forms.MessageBox.Show(ex.Message);
            }
        }

        private void buttonSelectEnt_Click(object sender, EventArgs e)
        {
            try
            {
                _AcEd.Editor editor = _AcAp.Application.DocumentManager.MdiActiveDocument.Editor;
                _AcEd.PromptEntityResult pres = editor.GetEntity("Select Entity");
                System.Windows.Forms.MessageBox.Show(pres.ToString());
            }
            catch
            {
                System.Windows.Forms.MessageBox.Show("Selection Failed");
            }
        }

    }
}

File: bbc/libs/BRXSDK_Bcad_V23_2_03/samples/VbBrxMgd/BrxMgdForm.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Runtime.InteropServices
Imports System.Windows.Forms

' ODA
Imports Teigha.Runtime
Imports Teigha.DatabaseServices
Imports Teigha.Geometry

' Bricsys
Imports Bricscad.ApplicationServices
Imports Bricscad.Runtime
Imports Bricscad.EditorInput


' com
Imports BricscadDb
Imports BricscadApp

' alias
Imports _AcRx = Teigha.Runtime
Imports _AcAp = Bricscad.ApplicationServices
Imports _AcDb = Teigha.DatabaseServices
Imports _AcGe = Teigha.Geometry
Imports _AcEd = Bricscad.EditorInput
Imports _AcGi = Teigha.GraphicsInterface
Imports _AcClr = Teigha.Colors
Imports _AcWnd = Bricscad.Windows

Namespace CsBrxMgd
    Partial Public Class BrxMgdForm
        Inherits Form
        Private m_document As AcadDocument
        Public Sub New(ByVal document As AcadDocument)
            If (document Is Nothing) Then
                Throw New ArgumentNullException("document == null")
            End If
            InitializeComponent()
            Me.m_document = document
        End Sub

        Private Sub buttonSelectEnt_Click(ByVal sender As Object, ByVal e As EventArgs) Handles buttonDist.Click
            Try

                MessageBox.Show(_AcAp.Application.DocumentManager.MdiActiveDocument.Editor.GetEntity("Select Entity").ToString)

            Catch ex As System.Exception
                MessageBox.Show("Selection Failed")
            End Try
        End Sub

        Private Sub select_Click(ByVal sender As Object, ByVal e As EventArgs) Handles buttonSelectEnt.Click
            Dim editor As Editor = _AcAp.Application.DocumentManager.MdiActiveDocument.Editor
            Try
                Dim pntOpt As New PromptPointOptions("Get First Point")
                Dim pntRes As PromptPointResult = editor.GetPoint(pntOpt)
                If (pntRes.Status <> PromptStatus.OK) Then
                    Throw New System.Exception("_AcEd.Editor.GetPoint failed")
                End If
                Dim distOpt As New PromptDistanceOptions("Get Next Point")
                distOpt.UseBasePoint = True
                distOpt.BasePoint = pntRes.Value
                Dim distRes As PromptDoubleResult = editor.GetDistance(distOpt)
                If (distRes.Status <> PromptStatus.OK) Then
                    Throw New System.Exception("_AcEd.Editor.GetDistance failed")
                End If
                MessageBox.Show(distRes.ToString)
            Catch ex As System.Exception
                MessageBox.Show(ex.Message)
            End Try
        End Sub

    End Class
End Namespace

